configure

```diff
@@ -343,17 +343,30 @@ def b(value):
 
 
 def pkg_config(pkg):
-  cmd = os.popen('pkg-config --libs %s' % pkg, 'r')
-  libs = cmd.readline().strip()
-  ret = cmd.close()
-  if (ret): return None
-
-  cmd = os.popen('pkg-config --cflags %s' % pkg, 'r')
-  cflags = cmd.readline().strip()
-  ret = cmd.close()
-  if (ret): return None
-
-  return (libs, cflags)
+  pkg_config = 'pkg-config'
+  pkg_config = os.environ.get('PKG_CONFIG', pkg_config)
+  pkg_config = os.environ.get('PKG_CONFIG_PATH', pkg_config)
+  args = '--silence-errors'
+  retval = ()
+  for flag in ['--libs-only-l', '--cflags-only-I', '--libs-only-L']:
+    try:
+      val = subprocess.check_output([pkg_config, args, flag, pkg])
+      # check_output returns bytes
+      val = val.encode().strip().rstrip('\n')
+    except subprocess.CalledProcessError:
+      # most likely missing a .pc-file
+      val = None
+    except OSError:
+      # no pkg-config/pkgconf installed
+      return (None, None, None)
+    retval += (val,)
+  return retval
+
+
+def format_libraries(list):
+  """Returns string of space separated libraries"""
+  set = list.split(',')
+  return ' '.join('-l{0}'.format(i) for i in set)
```

Comment:
Yeah, was mixed for readability (based on feedback by bnoordhuis). I previously had format everywhere.